### PR TITLE
test/unittest_lockdep: do not start extra threads

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by liewegas** [comment]
Oops, I merged too quickly, can you rebase this one?

**Comment by tchaikov** [comment]
@liewegas rebased and repushed.

File: src/test/common/test_lockdep.cc [hunk 1]
```diff
@@ -1,20 +1,45 @@
...
 #include "log/Log.h"
 
-TEST(lockdep, abba)
+class lockdep : public ::testing::Test
 {
-  ASSERT_TRUE(g_lockdep);
+protected:
+  void SetUp() override {
+#ifdef CEPH_DEBUG_MUTEX
+    GTEST_SKIP() << "WARNING: CEPH_DEBUG_MUTEX is not defined, lockdep will not work";
```
Comment (smithfarm) [inline src/test/common/test_lockdep.cc:20]: "CEPH_DEBUG_MUTEX is not defined" would seem to indicate `#ifndef` was intended.

see https://tracker.ceph.com/issues/46267

Comment (tchaikov) [reply src/test/common/test_lockdep.cc:20]: @smithfarm  thanks Nathan, should be fixed by #35856

**Pull request merged**